packages/react-core/src/filters/FilterTypes.js

```diff
@@ -50,8 +50,9 @@ function stringSearch(filterValues, featureValue, params = {}) {
   const normalizedFeatureValue = normalize(featureValue, params);
   const stringRegExp = filterValues
     .map((filterValue) => {
-      let stringRegExp = escapeRegExp(normalize(filterValue, params));
+      let stringRegExp = normalize(filterValue, params);
 
+      if (!params.useRegExp) stringRegExp = escapeRegExp(stringRegExp);
       if (params.mustStart) stringRegExp = `^${stringRegExp}`;
       if (params.mustEnd) stringRegExp = `${stringRegExp}$`;
```

Comment:
Should we normalize the string when using regexp?
Also, it's a bit strange to be able to use `mustStart`, `mustEnd`, `caseSensitive`… while using a regexp.
As an advanced user I would expect all these options to have no effect when using regexp, as I'm supposed to know how to write them as regexp only.

Reply:
Good catch @bbecquet. I agree with you (also Victor & Borja). We should ignore mustStart, mustEnd and caseSensitive. @juandjara

Reply:
Good catch @bbecquet I thought that too, but did not want to modify more of the existing feature. If you all agree, I'll add this change in a sec